networking_generic_switch/generic_switch_mech.py

```diff
@@ -77,19 +77,38 @@ def create_network_postcommit(self, context):
         provider_type = network['provider:network_type']
         segmentation_id = network['provider:segmentation_id']
         physnet = network['provider:physical_network']
+        created = []
+
+        def _rollback():
+            """Roll back a failed network creation."""
+            for switch_name, switch in created:
+                try:
+                    switch.del_network(segmentation_id, network_id)
```

Comment:
Do we assume that these networks won't exist then? Such that:

- network already exists
- call to create network succeeds
- another call to create network fails
- already existing network deleted

Reply:
That is a risk, yes. If you were to create a network and delete it then it would currently delete the VLAN.

I'm not sure what we can do about this - I'm not aware of any way to tell neutron that VLANs already exist. Potentially we could check the VLAN allocation range to see if this is a 'provider' VLAN or a 'tenant' VLAN, but I'm not sure that would be correct either.